app/code/core/Mage/Core/Helper/Data.php

```diff
@@ -147,31 +147,50 @@ public function formatPrice($price, $includeContainer = true)
     /**
      * Format date using current locale options and time zone.
      *
-     * @param   string|Zend_Date|null $date
+     * @param   string|Zend_Date|int|null $date If empty, return current local datetime.
      * @param   string              $format   See Mage_Core_Model_Locale::FORMAT_TYPE_* constants
      * @param   bool                $showTime Whether to include time
      * @return  string
      */
     public function formatDate($date = null, $format = Mage_Core_Model_Locale::FORMAT_TYPE_SHORT, $showTime = false)
     {
+        return $this->formatTimezoneDate($date, $format, $showTime);
+    }
+
+    /**
+     * Format date using current locale options and time zone.
+     *
+     * @param   string|Zend_Date|null $date   If empty, return current datetime.
+     * @param   string              $format   See Mage_Core_Model_Locale::FORMAT_TYPE_* constants
+     * @param   bool                $showTime Whether to include time
+     * @param   bool                $useTimezone Convert to local datetime?
+     * @return  string
+     */
+    public function formatTimezoneDate(
+        $date = null,
+        string $format = Mage_Core_Model_Locale::FORMAT_TYPE_SHORT,
+        bool $showTime = false,
+        bool $useTimezone = true
+    ) {
         if (!in_array($format, $this->_allowedFormats, true)) {
             return $date;
         }
-        if (!($date instanceof Zend_Date) && $date && !strtotime($date)) {
-            return '';
-        }
-        if (is_null($date)) {
-            $date = Mage::app()->getLocale()->date(Mage::getSingleton('core/date')->gmtTimestamp(), null, null);
-        } elseif (!$date instanceof Zend_Date) {
-            $date = Mage::app()->getLocale()->date(strtotime($date), null, null);
-        }
 
-        if ($showTime) {
-            $format = Mage::app()->getLocale()->getDateTimeFormat($format);
-        } else {
-            $format = Mage::app()->getLocale()->getDateFormat($format);
+        $locale = Mage::app()->getLocale();
+
+        if (empty($date)) {
+            $date = $locale->date(Mage::getSingleton('core/date')->gmtTimestamp(), null, null, $useTimezone);
+        } elseif (is_int($date)) {
+            $date = $locale->date($date, null, null, $useTimezone);
+        } elseif (!$date instanceof Zend_Date) {
+            $time = strtotime($date);
+            if ($time) {
+                $date = $locale->date($time, null, null, $useTimezone);
+            }
+            return '';
```

Comment:
<img width="551" alt="Screenshot 2023-05-23 alle 21 45 11" src="https://github.com/OpenMage/magento-lts/assets/909743/71840001-a221-4cd1-b46f-f3b0186770df">

this is for sure a mistake because both $time and $date are not used since it returns ''

Reply:
Right. It should be

```
} else {
    return '';
}
```

Reply:
that's more or less what I committed

Reply:
You removed that part ... `$date` is used in return statement.

```
            $time = strtotime($date);
            if ($time) {
                $date = $locale->date($time, null, null, $useTimezone);
            }
```

`if ($showTime) {` changed ...

This PR was NOT for `main` branch. #1525 was applied to `main` only cause of changing method signature. It should have been reverted first and replaced by this one. ATM this PRs looks not correct anymore.

Reply:
I understood that this PR was going to unify the same code on all branches, so main then ported to v19 and next, no?

Reply:
Its not on your screenshot, but `$date` is used in last line of that method. (`return $date->....`)

Reply:
I sencerely can't find that in the original diff https://github.com/OpenMage/magento-lts/files/11548215/2940.diff.txt it doesn't seem to me that that variable is needed

Reply:
Please check https://github.com/OpenMage/magento-lts/pull/1525/files and https://github.com/OpenMage/magento-lts/pull/2854/files.

I just missed that `else` statement.